tests-arquillian/src/test/java/org/jboss/weld/tests/producer/ejb/parameterized/ProducerDeclaredOnParameterizedSessionBeanViewTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2015, Red Hat, Inc. and/or its affiliates, and individual
+ * contributors by the @authors tag. See the copyright.txt in the
+ * distribution for a full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.weld.tests.producer.ejb.parameterized;
+
+import javax.inject.Inject;
+
+import junit.framework.Assert;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.weld.tests.category.Integration;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Kirill Gaevskii
+ */
+@RunWith(Arquillian.class)
+@Category(Integration.class)
+public class ProducerDeclaredOnParameterizedSessionBeanViewTest {
+
+    @Deployment
+    public static Archive<?> deployment() {
+        return ShrinkWrap.create(WebArchive.class).addPackage(ProducerDeclaredOnParameterizedSessionBeanViewTest.class.getPackage()).addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    @Inject
+    @Dummy
+    String injectedStringFromBusinessProducerMethod;
+
+    @Inject
+    @StaticDummy
+    String injectedStringFromStaticProducerMethod;
+
+    @Test
+    public void testProducerDeclaredOnParameterizedSessionBeanViewWithBusinessProducerMethod() {
+        Assert.assertEquals("hello", injectedStringFromBusinessProducerMethod);
+    }
+
+    @Test
+    //@Ignore("WELD-")
+    public void testProducerDeclaredOnParameterizedSessionBeanViewWithStaticProducerMethod() {
```

Comment:
The tested producer method is not declared on parameterized session bean view. Change the name of the test method accordingly.